ipa-core/src/protocol/ipa_prf/shuffle/mod.rs

```diff
@@ -43,6 +50,33 @@ where
         .collect::<Vec<_>>())
 }
 
+#[tracing::instrument(name = "shuffle_attribution_outputs", skip_all)]
+pub async fn shuffle_attribution_outputs<C, BK, TV, R>(
```

Comment:
If we don't try to merge BA just for the sake of shuffling (see comment below on attribution_outputs_to_shuffle) We do not need the generic R and the trait bounds. I don't think we should introduce new BA types just for the sake of shuffling.

Reply:
Yeah, I don't think these conversions to BA are necessary, in fact the sharded version of shuffle provides better ergonomics. I'm just using this version as a temporary stop-gap.